build/schema.md

```diff
@@ -0,0 +1,79 @@
+# Schema
+
+The value for `state` properties. Defines the default state property value, and its type.
+
+The schema definition follows a subset of [JSON Schema](https://json-schema.org/) but is more strict in terms of what properties are required.
+
+## Data types
+
+### enum
+
+
+The `enum` keyword is used to restrict a value to a fixed set of values. It must be an array with at least one element, where each element is unique.
+
+```json
+{  
+    "enum": ["red", "amber", "green"],
+    "default": "red"
+}
+```
+
+### number
+
+The `number` type is used for any numeric type, either integers or floating point numbers.
+
+```json
+{  
+    "type": "number",
+    "minimum": 0, // optional
```

Comment:
Are we sure this is what we want to start supporting? Schema parsing and validating is complicated, we are already doing it for the style json, I wouldn't want to introduce a different "language", if possible...

Reply:
Answered in https://github.com/maplibre/maplibre-style-spec/pull/1044#discussion_r2024346465